doc/references/taxonomy-parser.md

```diff
@@ -0,0 +1,161 @@
+   - *entry line*: it begins with `LC:`
+   - *stopwords line*: it begins with `stopwords:LC:`
+   - *global synonyms line*: it begins with `synonyms:LC`
+   - *property line*: it begins with a property_name
+     (we will use `PN` to mean any property name)
+     followed by a language code: `PN:LC:`
+
+     Note that *comment lines* are not *blank lines*.
+1. *meaningful lines* are : all lines type,
+   but *blank line* and *comment line*
+1. Each blank line begins a new block, but
+    1. each *stopwords line* or *global synonyms line*
+       should be considered a separated block,
+       (with preceding comment lines)
+    1. the first lines, up to the first *meaningful line*
+       (but preceding comment lines) is the *header*
+    1. other blocks must contains at least one *entry line*.
+       We call them *entry blocks*
+1. entry lines are composed of the `LC:` header,
+   then a list of *tags* separated by a comma `,`.
+   Comma without at least one space around are not separators.
+1. each *tag* as a *normalized version* referred to as *tagid*
+1. each *entry line* as a first tag which is referred to as *line tag*,
+   and as *line tagid* for it's *normalized version*.
+   *line tagid* is considered the identifier for this *entry line*,
+   and must be unique.
+1. the first *entry line* of each *entry block*
+   is referred to as *canonical tag* (resp. *canonical tagid*)
+   for this block.
+   This is considered the identifier for this block, and must be unique.
+1. the text in the *parent line* is composed of `LC:` followed by text.
+   This text is referred to as the *parent tag*
+
+## Sanitizing a line
+
+Each line we read in the file must be sanitized.
+
+This is the following process:
+
+1. remove any space characters at end of line
+   ```python
+   line = line.rstrip()
+   ```
+1. replace `’` (typographique quote) to simple quote `'`
+   ```python
+   line = line.replace("’", "'")
+   ```
+1. replace commas that have no space around by a lower comma character
+   and do the same for escaped comma (preceded by a `\`)
+   (to distinguish them from commas acting as tags separators)
+   ```python
+   line = re.sub(r"(\d),(\d)", r"\1‚\2", line)
+   line = re.sub(r"\\,", "\\‚", line)
+   ```
+1. removes parenthesis for roman numerals
+   ```python
+   line = re.sub(r"\(([ivx]+)\)", r"\1", line, flags=re.I)
+   ```
+
+## Normalized version of a string (of a tag)
+
+The *normalized version* of a string is the operation
+we will apply that to tag to obtain the tagid.
+
+It depends on the language.
+
+Obtain NFC form of the string. In python:
+```python
+string = unicodedata.normalize("NFC", my_input)
+```
+
+Then look at `string_normalization_for_lang` in `Config_off.pm` [^string_config]
+to know if following operations applies, based on language code:
+* lowercase
+* removing accents
+
+If lowercase applies and string is not a UUID, that is it does not match regexp `r"^[a-z\-]+\.[a-zA-Z0-9-_]{8}[a-zA-Z0-9-_]+$"`,
```

Comment:
Would you be able to elaborate on what a *UUID* is, with reference to this document?